indico/modules/receipts/schemas.py

```diff
@@ -0,0 +1,127 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2023 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+import yaml
+from marshmallow import ValidationError, fields, post_load, pre_load, validate, validates_schema
+from marshmallow_oneofschema import OneOfSchema
+
+from indico.core.marshmallow import mm
+from indico.modules.receipts.models.templates import ReceiptTemplate
+from indico.util.marshmallow import YAML, not_empty
+from indico.util.string import slugify
+
+
+class OwnerDataSchema(mm.Schema):
+    title = fields.String()
+    id = fields.Integer()
+    locator = fields.Dict()
+
+
+class _CheckboxAttributesSchema(mm.Schema):
+    value = fields.Bool()
+    label = fields.String(required=True)
+
+
+class _DropdownAttributesSchema(mm.Schema):
+    label = fields.String(required=True)
+    options = fields.List(fields.String())
+    default = fields.Int(validate=validate.Range(0))
+
+    @validates_schema(skip_on_field_errors=True)
+    def _validate_default(self, data, **kwargs):
+        if 'default' in data and data['default'] >= len(data['options']):
+            raise ValidationError('The provided value is out of range')
+
+
+class _InputAttributesSchema(mm.Schema):
+    value = fields.String()
+    label = fields.String(required=True)
+
+
+class CustomFieldAttributesSchema(OneOfSchema):
+    type_schemas = {
```

Comment:
A `textarea` field type would also be useful, e.g. for an address or similar multiline content